.github/workflows/npm-publish.yml

```diff
@@ -4,191 +4,238 @@ on:
     inputs:
       release_type:
-        description: 'Type of publish (one of: rc, main, both, hotfix, rc-hotfix)'
+        description: 'Type of publish (one of: rc, main, both, hotfix)'
         required: true
         default: 'rc'
-      force:
-        description: 'Force publish'
-        required: true
-        default: 'false'
-      desc:
-        description: 'Release description'
+      branch:
+        description: 'Hotfix branch'
         required: false
-        default: ''
+
+env:
+  # Dagger
+  DAGGER_PLAN: cue.mod/pkg/github.com/3box/pipeline-tools/ci/plans/ceramic.cue
+  DAGGER_LOG_FORMAT: "plain"
+  PIPELINE_TOOLS_VER: "0.2.0"
+  # Secrets
+  AWS_ACCOUNT_ID: ${{ secrets.AWS_ACCOUNT_ID }}
+  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+  AWS_REGION: ${{ secrets.AWS_DEFAULT_REGION }}
+  GH_TOKEN: ${{ secrets.GH_TOKEN }}
+  NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
 
 jobs:
   publish_main:
     runs-on: ubuntu-latest
+    if: ${{ github.event.inputs.release_type == 'both' || github.event.inputs.release_type == 'main' }}
     steps:
-      - name: Setup environment
-        id: set-env
-        run: |
-          PUBLISH_ARGS=""
-          if [[ "${{ github.event.inputs.release_type }}" == "both" || "${{ github.event.inputs.release_type }}" == "main" ]]; then
-            echo "::set-output name=main::true"
-            PUBLISH_ARGS="$PUBLISH_ARGS --conventional-graduate"
-          elif [[ "${{ github.event.inputs.release_type }}" == "hotfix" ]]; then
-            echo "::set-output name=main::true"
-            echo "::set-output name=hotfix::true"
-          fi
-          if [[ "${{ github.event.inputs.force }}" == "true" ]]; then
-            PUBLISH_ARGS="$PUBLISH_ARGS --force-publish"
-          fi
-          if [[ -n $PUBLISH_ARGS ]]; then
-            echo "::set-output name=publishArgs::-- $PUBLISH_ARGS"
-          fi
-      - name: Setup Node
-        if: ${{ steps.set-env.outputs.main == 'true' }}
+      -
+        name: Setup Node
         uses: actions/setup-node@v1
         with:
           node-version: 16
-      - name: Checkout `main` branch
-        if: ${{ steps.set-env.outputs.main == 'true' }}
+      -
+        name: Checkout main branch
         uses: actions/checkout@v2
         with:
-          token: ${{ secrets.GH_TOKEN }}
           ref: main
-      - name: Configure git
-        if: ${{ steps.set-env.outputs.main == 'true' }}
+      -
+        name: Configure git
         run: |
           git config user.email "${{ github.actor }}@users.noreply.github.com"
           git config user.name ${{ github.actor }}
           git fetch --all
-      - name: Merge down from `main` -> `rc` -> `develop`
-        if: ${{ steps.set-env.outputs.main == 'true' }}
+      -
+        name: Merge down from main -> rc -> develop
         run: |
           git checkout -B release-candidate refs/remotes/origin/release-candidate
           git merge main
```

Comment:
Do we need a `git push` step here?  We have it below for the post-release merges.

Reply:
No, not needed here. The `publish` command will automatically push.

Reply:
but the publish happens on main, here we're merging to the release-candidate and develop branches

Reply:
Oh wait, I thought you were looking at a different spot. I intentionally left push out from this spot to avoid extra commits since each commit comes with CircleCI tests + image verification + deploys (since this happens on deployable branches).

This increases the downtime from redeploys when we could just wait a few more steps and then push.

Though, now that I'm thinking about it, each commit, even pushed together with other later, will still do all these steps :( The one advantage of waiting till the end is that the CD manager is likely to collapse multiple back-to-back deploys.

Not a huge advantage with our current workflows, I guess, but at least now you know the idea behind doing it this way :)